src/traces/mesh3d/convert.js

```diff
@@ -105,6 +134,18 @@ proto.update = function(data) {
 
     var cells;
     if(data.i && data.j && data.k) {
+
+        if(
+            data.i.length !== data.j.length ||
+            data.j.length !== data.k.length ||
+            !hasValidIndices(data.i, numVertices) ||
+            !hasValidIndices(data.j, numVertices) ||
+            !hasValidIndices(data.k, numVertices) ||
+            !isTriangle([data.i, data.j, data.k])
+        ) {
+            data.visible = false;
```

Comment:
We can't set `visible: false` here. `fullData` keys can't be mutated after the defaults step (unless they are `_` properties) for our `Plotly.react` diffing to work correctly.

Just removing this line here will suffice (i.e. and keeping the early return below) should suffice.